components/function-controller/design/extend_function_spec_proposal.md

```diff
@@ -0,0 +1,529 @@
+
+Cons: 
+ - Not allows to mount anything besides secrets
+
+A) and B) are not exclusive
+
+We could separate those cases. (See last 'compromise' option)
+
+
+### Metadata for function pod
+
+Define runtime labels and annotations on the root level or under a `metadata` field
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    ...
+  annotations:
+    ...
+spec:
+  metadata: 
+    labels: ...
+    annotations: ...
+```
+OR
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    ...
+  annotations:
+    ...
+spec:
+  labels: ...
+  annotations: ...
+```
+
+## Samples: 
+
+### Option 1
+
+- simplified mounts serving just for service binding purpose
+- stretch: volume mounts could be added as a separate feature
+- extract `build` object for any build-time specific config
+- runtime labels and annotations on the spec root level
+
+```yaml
+apiVersion: serverless.kyma-project.io/v1alpha2
+kind: Function
+metadata:
+  name: my-function
+  namespace: default
+  labels:
+    app.kubernetes.io/name: my-function
+spec:
+  runtime: nodejs16
+  source:
+    ...
+  secretBindings: 
+  - source: my-secret
+    mountPath: "/foo" # optional.. read from SERVICE_BINDING_ROOT ENV 
+
+  profile: S / M / L / XL / ... / Custom #optional
+  resources: # optional... required if spec.profile==Custom
+    limits: ... #if_custom
+    requests: ... #if_custom
+​  
+  labels: 
+    app: my-app
+  annotations: 
+    fluentbit.io/parser: my-regex-parser
+    istio-injection: enabled
+​
+  env:
+  - name: SERVICE_BINDING_ROOT
+    value: /service_bindings
+​​
+  build:
+    profile: S / M / L / XL / ... / Custom
+    resources: # optional... required if spec.build.profile==Custom
+      limits: ... #if_custom
+      requests: ... #if_custom
+    labels: #optional
+    annotations: #optional
+```
+
+### Option 2
+
+- almost the same as `Option 1` (same pros)
+- dif1: move some fields from `.spec` to the new struct `.spec.runtimeSpec` to clearly distinguish fields desired to be used in the `build` and `running` phases. For example in `Option 1` users may have questions after seeing `.spec.env` and `.spec.build.env` fields for example "is .spec.env dedicated for the running function's pod? Would the field be merged with .spec.build.env for the building job?"
+- dif2: rename the `.spec.profile` to the `.spec.resourcesProfile` to make this field more intuitive
+- dif3: this solution is simple but not as simple as `Option 1`
+
+>NOTE: the main idea is to close a specific configuration in a field that represents the specific phase of the function's lifecycle. It would be intuitive and easy to understand for a user that the `.spec.build` field contains configuration for the building phase and `.spec.runtimeSpec` for the running phase.
```

Comment:
```suggestion
>NOTE: the main idea is to close a specific configuration in a field that represents the specific phase of the Function's lifecycle. It would be intuitive and easy to understand for a user that the `.spec.build` field contains configuration for the building phase and `.spec.runtimeSpec` for the running phase.
```